velox/common/memory/tests/SharedArbitratorTest.cpp

```diff
@@ -2747,6 +2747,61 @@ DEBUG_ONLY_TEST_F(SharedArbitrationTest, arbitrateMemoryFromOtherOperator) {
   }
 }
 
+TEST_F(SharedArbitrationTest, joinBuildSpillError) {
+  const int kMemoryCapacity = 32 << 20;
+  // Set a small memory capacity to trigger hash join spill.
+  setupMemory(kMemoryCapacity, 0);
+
+  const int numVectors = 16;
```

Comment:
Do we need 16 vectors? How did you choose this number, 16?

Reply:
No particular reason but have tried the other number. Basically the test needs 32MB memory usage.

Reply:
> Basically the test needs 32MB memory usage.

Wondering why is this? Maybe add a comment to clarify? Other readers may have the same question. What happens if you use fewer vectors? Does the test fail?

Reply:
Is this because kMemoryCapacity is 32MB? Would it make sense to add a helper function that takes number of bytes and produces vectors that use at least that much memory?

Reply:
That's good point that we could consider to have some utility to leverage estimate size for that. I will do that in a followup.